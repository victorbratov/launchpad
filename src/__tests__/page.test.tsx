import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '../app/page'

// test for sign up button
describe('Page', () => {
  it('has a sign up button', () => {
    render(<Page />)
 
    expect(screen.getByRole('link', { name: 'Sign Up' })).toHaveAttribute('href', '/signup')
  })
})

// test for log in button
describe('Page', () => {
  it('has a log in button', () => {
    render(<Page />)
    expect(screen.getByRole('link', { name: 'Log In' })).toHaveAttribute('href', '/sign-in')
  })
})

// Snapshot test to check for any unepxected UI changes
test("renders welcome page", () => {
  render(<Page />);
  const { container } = render(<Page />);
  expect(container).toMatchSnapshot();
});