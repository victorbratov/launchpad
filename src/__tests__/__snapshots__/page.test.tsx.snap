// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders welcome page 1`] = `
<div>
  <div
    class="font-mono grid grid-rows-[20px_1fr_20px] items-center justify-items-center min-h-screen p-8 pb-20 gap-16 sm:p-20"
  >
    <header>
      <div
        class="w-full items-center text-4xl"
      >
        <h1>
          Launchpad
        </h1>
      </div>
    </header>
    <main
      class="flex flex-col gap-[32px] row-start-2 items-center"
    >
      <div
        class="flex flex-col gap-[32px] row-start-2 items-center"
      >
        <h1
          class="text-center"
        >
          Welcome to Launchpad
        </h1>
        <p
          class="w-1/2 text-center"
        >
          Launchpad is an investment platform that allows businesses and investors to collaborate openly and effectively. 
        </p>
        <p
          class="w-1/2 text-center"
        >
           Log in to a business account to receive funding from investors. Log in to an investor account to support business ideas and gain from the profits.
        </p>
      </div>
      <div
        class="flex gap-4 items-center flex-col sm:flex-row"
      >
        <a
          class="rounded-full border border-solid border-transparent transition-colors flex items-center justify-center bg-foreground text-background gap-2 hover:bg-[#383838] dark:hover:bg-[#ccc] font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 sm:w-auto"
          href="/sign-in"
          rel="noopener noreferrer"
          target="_blank"
        >
          Log In
        </a>
        <a
          class="rounded-full border border-solid border-black/[.08] dark:border-white/[.145] transition-colors flex items-center justify-center hover:bg-[#f2f2f2] dark:hover:bg-[#1a1a1a] hover:border-transparent font-medium text-sm sm:text-base h-10 sm:h-12 px-4 sm:px-5 w-full sm:w-auto md:w-[158px]"
          href="/signup"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sign Up
        </a>
      </div>
    </main>
  </div>
</div>
`;
